transfermethodui/src/androidTest/java/com/hyperwallet/android/ui/transfermethod/insight/SelectTransferMethodInsightTest.java

```diff
@@ -0,0 +1,159 @@
+    public static HyperwalletExternalResourceManager sResourceManager = new HyperwalletExternalResourceManager();
+
+    @Rule
+    public HyperwalletSdkRule mHyperwalletSdkRule = new HyperwalletSdkRule();
+    @Rule
+    public HyperwalletInsightMockRule mHyperwalletInsightMockRule = new HyperwalletInsightMockRule();
+    @Rule
+    public HyperwalletMockWebServer mMockWebServer = new HyperwalletMockWebServer(8080);
+    @Rule
+    public ActivityTestRule<SelectTransferMethodActivity> mActivityTestRule =
+            new ActivityTestRule<>(SelectTransferMethodActivity.class, true, false);
+    @Rule
+    public MockitoRule mMockito = MockitoJUnit.rule();
+    @Captor
+    private ArgumentCaptor<Map<String, String>> mParamsCaptor;
+
+    @Before
+    public void setup() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("authentication_token_response.json")).mock();
+        IdlingRegistry.getInstance().register(EspressoIdlingResource.getIdlingResource());
+    }
+
+    @After
+    public void cleanup() {
+        TransferMethodRepositoryFactory.clearInstance();
+        UserRepositoryFactory.clearInstance();
+        IdlingRegistry.getInstance().unregister(EspressoIdlingResource.getIdlingResource());
+    }
+
+    @Test
+    public void testSelectTransferMethod_verifyInsightEventCreatedOnTransferMethodOptionsLoad() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("user_response.json")).mock();
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("successful_tmc_keys_response.json")).mock();
+
+        mActivityTestRule.launchActivity(null);
+
+        verify(mHyperwalletInsightMockRule.getInsight(),
+                timeout(5000).times(1)).trackImpression(any(Context.class),
+                eq(HyperwalletInsight.PAGE_TRANSFER_METHOD_SELECT),
+                eq(HyperwalletInsight.TRANSFER_METHOD_GROUP),
+                mParamsCaptor.capture());
+
+        assertThat(mParamsCaptor.getValue().get("hyperwallet_ea_country"), is("US"));
+        assertThat(mParamsCaptor.getValue().get("hyperwallet_ea_currency"), is("USD"));
+    }
+
+    @Test
+    public void testSelectTransferMethod_verifyInsightEventCreatedOnCountrySelection() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("user_response.json")).mock();
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("successful_tmc_keys_response.json")).mock();
+
+        mActivityTestRule.launchActivity(null);
+
+        verify(mHyperwalletInsightMockRule.getInsight(),
+                times(1)).trackImpression(any(Context.class),
+                eq(HyperwalletInsight.PAGE_TRANSFER_METHOD_SELECT),
```

Comment:
Variable Name and the class has been changed for these Strings

Reply:
Updated thanks